Scripts/Modules/WorkspaceModule.cs

```diff
@@ -51,12 +52,20 @@ class WorkspaceLayout
 		internal List<IWorkspace> workspaces { get { return m_Workspaces; } }
 		readonly List<IWorkspace> m_Workspaces = new List<IWorkspace>();
 
+		internal List<WorkspaceInput> workspaceInputs { get { return m_WorkspaceInputs; } }
+		readonly List<WorkspaceInput> m_WorkspaceInputs = new List<WorkspaceInput>();
+
 		internal event Action<IWorkspace> workspaceCreated;
 		internal event Action<IWorkspace> workspaceDestroyed;
 
 		internal static List<Type> workspaceTypes { get; private set; }
 
-		public bool preserveWorkspaces { get; set; }
+		internal Transform leftRayOrigin { private get; set; }
+		internal Transform rightRayOrigin { private get; set; }
+
+		internal bool preserveWorkspaces { get; set; }
+
+		internal Func<Transform, float> getPointerLength { private get; set; }
```

Comment:
Sure! I've also found myself needing this in lots of places. I'll make a GetPointerLength,  but there could be arguments for just adding it to IUsesDirectSelection.